### DISCUSSION TITLE
Inoreader 加星后自动保存到简悦的稍后读以及生成本地快照

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3007] [服务]
> 如果你是 Inoreader 用户的话，试想一种完美的信息过滤方案：

将 Inoreader 的某个条目加星操作，然后这个 URL 对应的快照就会出现在你坚果云的相应文件夹中，同时也会出现在简悦的稍后读中，而后方便你在稍后读中对它进行二次加工。

https://user-images.githubusercontent.com/81074/140274712-dda358a5-2b81-4f97-ac7d-8f87cd58c145.mp4

# 前置知识

> 需要事先完成下面的配置。

1. [如何开通开放平台](https://github.com/Kenshin/simpread/discussions/2757)
2. [API 快照功能](https://github.com/Kenshin/simpread/discussions/2994)
3. [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)

# 注册 integrately

> 如果你用过 IFTTT / Zapier 的话，integrately 是一个比 IFTTT 更强大的自动化程序，更难能可贵的是，免费版的即可使用 Webhook 服务。（ _Zapier 的 Webhook 收费，虽然 IFTTT 不收费，但它的定制化无法跟 integrately 相比_）

> 如果没有账户的话，请使用 **[此链接](https://integrately.com?ref=62pcvo)** 注册，这样你和我都可以**增加 500个 Task**

# 获取 Token

> 获取简悦的 Token 地址，登录 [开放平台](http://simpread.pro/developer) 然后复制你的 Token

> 开通方式 [请看这里](https://github.com/Kenshin/simpread/discussions/2757)

![image](https://user-images.githubusercontent.com/81074/140269640-b0d4e509-521b-4155-a1b3-d4eb6e6b60e5.png)

> 将下面代码中的 `你的 Token` 替换为 `刚才复制的内容`

```
https://api-wrap.simpread.pro/api/service/webhook/<你的 Token>
```

# 配置

>  连接 Webhook 与 Inoreader

![image](https://user-images.githubusercontent.com/81074/140268331-91ef3224-8e7b-4d14-a3fb-483eeb0e658b.png)

> 配置 Webhook 与 Inoreader

![image](https://user-images.githubusercontent.com/81074/140269344-7c886f36-00cd-4d4f-acf5-d29a32e028a7.png)

> 测试并完成当前设置

![image](https://user-images.githubusercontent.com/81074/140270917-a3f856d7-f520-40b2-82b0-ebbb3b5419d8.png)

![image](https://user-images.githubusercontent.com/81074/140271047-fff24e3c-4103-4d52-8a20-9f72468bc187.png)

# 使用场景

因调用 integrately 的自动化服务有一定时间差，所以当你加星后并不能马上实时产生自动化，所以最好的使用场景：

> 在手机上使用 Inoreader 时，看到需要深度阅读 / 标注 / 需要留存的文章，使用此方式。同样此手段也可以作为一种有效的信息降噪方案。

# 注意

1. 仅配置 `url` 即可，这样才会生成 URL 的快照

2. 如果在配置时出现下图的提示，**请使用国内的 RSS 源** e.g. [少数派](https://sspai.com/feed) 来作为测试使用。

![image](https://user-images.githubusercontent.com/81074/140468077-5eadd9fe-8238-469f-affd-299251ab0ac4.png)

# 关联

- [利用 Airtable 实现属于自己的链接公开分享集 #3003](https://github.com/Kenshin/simpread/discussions/3003)
- [Raindrop.io 加星后自动导入到简悦并生成本地快照 #3021](https://github.com/Kenshin/simpread/discussions/3021)
- [加入 Pocket 后自动保存到简悦的稍后读（包含快照） #3024](https://github.com/Kenshin/simpread/discussions/3024)
- [Instapaper 加星后自动保存到简悦的稍后读（包含快照） #3034](https://github.com/Kenshin/simpread/discussions/3034)
- [将简悦变成你的 RSS 阅读器（自动下载快照到本地） #3036](https://github.com/Kenshin/simpread/discussions/3036)
- [保存到 Pinboard.in 同时也保存到简悦（包括本地快照） #3048](https://github.com/Kenshin/simpread/discussions/3048)
- [订阅 Newsletter 后自动保存到简悦的稍后读 #3088](https://github.com/Kenshin/simpread/discussions/3088)
- [当加入稍后读后，自动发送到你的 Telegram Channel #3092](https://github.com/Kenshin/simpread/discussions/3092)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）#3104](https://github.com/Kenshin/simpread/discussions/3104)
- [保存 RSS 的全文到简悦的稍后读（包括快照） #3186](https://github.com/Kenshin/simpread/discussions/3186)

# 引申

- [[教程\]如何开通开放平台 #2757](https://github.com/Kenshin/simpread/discussions/2757)
- [当在开放平台设置完毕后，如何测试配置是否成功？ #3020](https://github.com/Kenshin/simpread/discussions/3020)
- [授权登录（如何使用第三方登录简悦的高级账户） #1570](https://github.com/Kenshin/simpread/discussions/1570)
- [个人中心 #2959](https://github.com/Kenshin/simpread/discussions/2959)
- [免密码登录 #2974](https://github.com/Kenshin/simpread/discussions/2974)

**Comment by firstuanl** [comment]
UP

**Reply by weihaoxie** [reply]
不知道为啥，每次加星之后很久才被加入到稍后读，有些还漏了

**Reply by Kenshin** [reply]
> 不知道为啥，每次加星之后很久才被加入到稍后读，有些还漏了

Integrately task 并不是实时的，免费版估计是 15 分钟左右。另外，也并不是  100% 都能保存成功，当调用失败时，应该会给你的邮箱（ Integrately 账户）发送一个调用失败的邮件。

如果 URL 在正常网络环境下无法访问，则也无法生成快照。

**Comment by chenggiant** [comment]
每一次加星，就用掉一个Integrately task么？

**Reply by Kenshin** [reply]
是的，所以一般来说，只有加星的内容才值得被收藏。

PS：你可以使用它的邀请链接注册机制，每邀请一个增加 500 个 task

**Comment by Simonup** [comment]
请问出现deployment function invocation timeout 是因为国外网站无法保存吗？

**Reply by Kenshin** [reply]
是的，快照是生成在阿里云里面（因为速度快），后续会考虑放在其它服务中，就能解决你的问题了。

**Comment by scramjetFoam** [comment]
网站的配置流程变化了，如果可以的话，希望您能更新一下教程。

**Reply by Kenshin** [reply]
什么「网站的配置流程变化了」？

**Reply by scramjetFoam** [reply]
integrately，但是我又发现也就是第一步选择post有点不一样，点进去之后下一步很好理解，影响不大。

**Reply by Kenshin** [reply]
okay 不过建议最好能截图指出。

**Comment by hellozgl** [comment]
telegram bot的链接能自动离线下载html文件
通过收藏助手也能自动离线下载html文件
收藏到instapaper并打星标的文章，通过IFTTT,也可以自动离线下载html文件（效率很高，基本做到几秒以内）
但通过这个integrately，在inoreader打星后，（十五分钟以后）会更新到稍后读列表，但是却没有自动在坚果云生成离线html文件。测试链接是可以国内正常访问的网站，例如：https://www.taoguba.com.cn/Article/3681373/1  
这是因为integrately不给离线下载html的功能了吗？不知还有否其他类似的关联工具可以用于inoreader使用。

![屏幕截图 2022-02-11 14 39 52](https://user-images.githubusercontent.com/70676660/153547787-ef962028-fd0d-4a65-8744-c59f5de5419f.png)
![屏幕截图 2022-02-11 14 40 09](https://user-images.githubusercontent.com/70676660/153547791-77bc02f4-24f3-4d06-8916-b295bf674179.png)

**Reply by Kenshin** [reply]
> 通过这个integrately，在inoreader打星后，（十五分钟以后）会更新到稍后读列表，但是却没有自动在坚果云生成离线html文件。

15分钟左右是没问题的（也可能会稍微快点），至于快照的问题，是不是下面这个引起的？

![image](https://user-images.githubusercontent.com/81074/153549533-244a0a71-c72c-4ab1-9f86-7beb869b2df2.png)

**Reply by Kenshin** [reply]
> 这是因为integrately不给离线下载html的功能了吗？不知还有否其他类似的关联工具可以用于inoreader使用。

跟这个没关系的，离线 HTML 是简悦的功能。

**Reply by hellozgl** [reply]
我只配置了url，如下图圆圈所示：
![屏幕截图 2022-02-11 14 39 52](https://user-images.githubusercontent.com/70676660/153550711-00311663-6d8e-465f-921e-849d014bdaf2.png)

**Reply by Kenshin** [reply]
而对于这个奇怪的网页：https://www.taoguba.com.cn/Article/3681373/1
使用telegram bot 和 收藏助手的效果一致，都是能推送到稍后读，含标题，但是都无法自动下载html。如果使用innoreader+星标+integrately的话，效果更差，连标题都没了，更别提自动下载html到坚果云了。
![屏幕截图 2022-02-11 16 02 41](https://user-images.githubusercontent.com/70676660/153556578-e796a122-5a70-4779-902f-16f31c64ac95.png)

via @[hellozgl](https://github.com/hellozgl)

**Reply by Kenshin** [reply]
嗨 不要在两个 discussions 讨论相同的内容...  😂 

我统一在这里回复你：

1. 快照不是 100% 能生成成功，你可以理解为 简悦的服务器爬虫去爬页面，但某些页面有可能阻止爬虫。
2. 同步助手在国内，只能保存国内的内容，同理 Telegram bot

另外，可以测试下正常点的页面 e.g https://sspai.com/post/71200 看看能否生成快照，一般来说 Telegram bot 保存方式要更好些。（因为是自己写的，而这些自动化服务都是第三方方案，肯定不如自己写的快。

**Reply by hellozgl** [reply]
明白了。那看来以后优先级还是以telegram bot为主。少数派这个网页都能正常快照。
但可以试试这个网站https://www.nytimes.com/zh-hans/2022/02/10/world/asia/russia-ukraine-drill.html
使用telegram可以生成快照。使用收藏助手就没办法生成快照了。

**Reply by Kenshin** [reply]
> 但可以试试这个网站https://www.nytimes.com/zh-hans/2022/02/10/world/asia/russia-ukraine-drill.html
使用telegram可以生成快照。使用收藏助手就没办法生成快照了。

这是正常的，下一步我会考虑把同步助手也支持 Telegram 一样的效果。

**Comment by sausea** [comment]
hi，Kenshin 非常感谢做出简悦这么棒的产品，但是国外网站的RSS功能采用这种方法做信息聚合太繁琐了，尤其像nber、economitrica这种网站适配性也存在一定的问题，有没有考虑过直接将RSS跟稍后读整合一下？

**Reply by Kenshin** [reply]
谢谢喜欢简悦。

RSS 的导入已经有了，你可以看 https://github.com/Kenshin/simpread/discussions/3186

当然并不是原生功能，还是借助 API 的方案。

**Comment by woshishui10000** [comment]
<img width="555" height="191" alt="image" src="https://github.com/user-attachments/assets/5d9a3d1a-e137-48a2-a288-d5ad935e2875" />

不知道是什么问题？

**Reply by Kenshin** [reply]
嗨，一般来说配置时遇到了问题，需要详细描述配置过程（截图）才能定位到错误。

PS：这个功能已经发布了有一段时间了。

**Reply by woshishui10000** [reply]
那好吧。过两天再从头捋一遍看看。没有代码方面基础，干起来太难了😭

**Reply by Kenshin** [reply]
如果在 Web 端使用的话，其实简悦已经适配了 Inoreader 可以直接进入阅读模式。

这种方式效果最好。
